### PR TITLE
Add type hints to files under integration_tests

### PR DESCRIPTION
**Pull request by gasin**
## Motivation
Please refer to the Issue #711 .

## Description of the changes
I fixed the type annotations of files under `tests/integration_tests`

### CODE REVIEW

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your update. I have a comment on the removed assertion.

File: tests/integration_tests/lightgbm_tuner_tests/test_optimize.py [hunk 1]
```diff
@@ -399,11 +384,6 @@ def test_sample_train_set(self):
         )
         runner.sample_train_set()
 
-        # Workaround for mypy.
-        if not type_checking.TYPE_CHECKING:
-            runner.train_subset.construct()  # Cannot get label before construct `lgb.Dataset`.
-            assert runner.train_subset.get_label().shape[0] == sample_size
-
```
[the remark this answers is not in the document]
Comment (gasin) [reply tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:406]: At the beggining of files, we treated `type_cheking.TYPE_CHECKING` as always being true, so I think `not type_chking.TYPE_CHKING` should be always false.
Is this wrong?

[the remark this answers is not in the document]
Comment (toshihikoyanase) [reply tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:406]: > `not type_chking.TYPE_CHKING` should be always false.

In my understanding, `type_chking.TYPE_CHKING` is `True` when `mypy` checks the code, and it is `False` when Python interpreters execute the code. The comment L402 suggested L404 and L405 do not work with `mypy`, but they work with `pytest`.

[the remark this answers is not in the document]
Comment (gasin) [reply tests/integration_tests/lightgbm_tuner_tests/test_optimize.py]: Thank you for the concise explanation.
Restored.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM.

@harupy I think your suggestions were included in this PR. If you have further comments, please let me know. Otherwise, I'll merge it since it received two approvals.

**Review by harupy — APPROVED** [review]
lgtm

**Comment by toshihikoyanase** [comment]
@harupy Thank you for your quick response. Let me merge this PR.

**Pull request merged**